appsec/src/helper/remote_config/client_handler.cpp

```diff
@@ -57,13 +65,36 @@ std::unique_ptr<client_handler> client_handler::from_settings(
         remote_config::client::from_settings(rc_settings, std::move(listeners));
 
     return std::make_unique<client_handler>(
-        std::move(rc_client), std::move(service_config));
+        std::move(rc_client), std::move(service_config), std::move(msubmitter));
 }
 
 void client_handler::poll()
 {
+    const std::lock_guard lock{mutex_};
+
     try {
-        rc_client_->poll();
+        if (last_success_ != empty_time) {
+            auto now = std::chrono::steady_clock::now();
+            auto elapsed =
+                std::chrono::duration_cast<std::chrono::milliseconds>(
+                    now - last_success_);
+            msubmitter_->submit_metric(metrics::rc_last_success,
+                static_cast<double>(elapsed.count()), {});
+        }
+
+        const bool result = rc_client_->poll();
+
+        auto now = std::chrono::steady_clock::now();
+        last_success_ = now;
+
+        if (result && creation_time_ != empty_time) {
+            auto elapsed =
+                std::chrono::duration_cast<std::chrono::milliseconds>(
+                    now - creation_time_);
+            msubmitter_->submit_metric(metrics::rc_first_pull,
```

Comment:
Interesting, you added the only metrics no one else has implemented (`appsec.remote_config`) and which I was deprecating... I might suggest removing them for now as, if they're not removed, the namespace will change from `appsec` to something else.